src/pocket/PocketECSCodePipeline.ts

```diff
@@ -33,115 +45,64 @@ export class PocketECSCodePipeline extends Resource {
   ) {
     super(scope, name);
 
+    this.codeBuildProjectName = this.getCodeBuildProjectName();
+    this.codeDeployApplicationName = this.getCodeDeployApplicationName();
+    this.codeDeployDeploymentGroupName =
+      this.getCodeDeployDeploymentGroupName();
+    this.taskDefinitionTemplatePath = this.getTaskDefinitionTemplatePath();
+    this.appSpecTemplatePath = this.getAppSpecTemplatePath();
+
+    this.s3KmsAlias = this.createS3KmsAlias();
+    this.pipelineArtifactBucket = this.createArtifactBucket();
+    this.pipelineRole = this.createPipelineRole();
     this.codePipeline = this.createCodePipeline();
   }
 
-  /**
-   * Create a CodePipeline that runs CodeBuild and ECS CodeDeploy
-   * @private
-   */
-  private createCodePipeline(): codepipeline.Codepipeline {
-    const pipelineRole = this.createPipelineRole();
+  private getPipelineName = () => `${this.config.prefix}-CodePipeline`;
 
-    const s3KmsAlias = new kms.DataAwsKmsAlias(this, 'kms_s3_alias', {
-      name: 'alias/aws/s3',
-    });
+  private getCodeBuildProjectName = () =>
+    this.config.codeBuildProjectName ?? this.config.prefix;
 
-    const pipelineArtifactBucket = this.createArtifactBucket();
+  private getCodeDeployApplicationName = () =>
+    this.config.codeDeploy?.applicationName ?? `${this.config.prefix}-ECS`;
 
-    const codeBuildProjectName =
-      this.config.codeBuildProjectName ?? this.config.prefix;
+  private getCodeDeployDeploymentGroupName = () =>
+    this.config.codeDeploy?.deploymentGroupName ?? `${this.config.prefix}-ECS`;
 
-    const codeDeployApplicationName =
-      this.config.codeDeploy?.applicationName ?? `${this.config.prefix}-ECS`;
+  private getTaskDefinitionTemplatePath = () =>
+    this.config.codeDeploy?.taskDefPath ??
+    PocketECSCodePipeline.DEFAULT_TASKDEF_PATH;
 
-    const codeDeployDeploymentGroupName =
-      this.config.codeDeploy?.deploymentGroupName ??
-      `${this.config.prefix}-ECS`;
+  private getAppSpecTemplatePath = () =>
+    this.config.codeDeploy?.appSpecPath ??
+    PocketECSCodePipeline.DEFAULT_APPSPEC_PATH;
 
-    this.attachPipelineRolePolicy(
-      pipelineRole,
-      pipelineArtifactBucket,
-      codeBuildProjectName,
-      codeDeployApplicationName,
-      codeDeployDeploymentGroupName
-    );
+  /**
+   * Get all stages for the pipeline, including postDeployStage if provided.
+   * @private
+   */
+  private getStages = () => [
+    this.getSourceStage(),
+    this.getDeployStage(),
+    ...(this.config.postDeployStage ? [this.config.postDeployStage] : []),
```

Comment:
what do you think about making `config.postDeployStage` an array? (perhaps with a rename to `postDeployStages`.) would make it a little more extensible and simplify this line of code as well.